runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -22,14 +22,21 @@ use crate::{actions::execute_function_call, ext::RuntimeExt};
 
 pub mod errors;
 
-pub struct TrieViewer {}
+pub struct TrieViewer {
+    /// Upper bound of the byte size of contract state that is still viewable. None is no limit
+    state_size_limit: Option<u64>,
+}
 
 impl TrieViewer {
-    /// Upper bound of the size of contract state that is still viewable.
-    const CONTRACT_STATE_SIZE_LIMIT: u64 = 50_000;
+    pub const DEFAULT_CONTRACT_STATE_SIZE_LIMIT: Option<u64> = Some(50_000);
 
     pub fn new() -> Self {
-        Self {}
+        Self { state_size_limit: Self::DEFAULT_CONTRACT_STATE_SIZE_LIMIT }
+    }
+
+    pub fn state_size_limit(mut self, limit: Option<u64>) -> Self {
+        self.state_size_limit = limit;
+        self
     }
```

Comment:
It feels a bit odd to have `DEFAULT_CONTRACT_STATE_SIZE_LIMIT` be specified here. In the only "real" use-case, we immediately override the default value:

https://github.com/near/nearcore/pull/4199/files#diff-f0e8081b926da5d399c24bd42912c9cf2cb268e2d783b4f5f3184ecfbec33ee1R150

Logically, defaults belong to config as well. 

I think (70% confidence) that it's better to change 

```rust
impl TrieViewer {
    pub const DEFAULT_CONTRACT_STATE_SIZE_LIMIT: Option<u64> = Some(50_000);

    pub fn new() -> Self {
        Self { state_size_limit: Self::DEFAULT_CONTRACT_STATE_SIZE_LIMIT }
    }

    pub fn state_size_limit(mut self, limit: Option<u64>) -> Self {
        self.state_size_limit = limit;
        self
    }
}
```

to

```rust
impl TrieViewer {
    pub fn new_with_state_size_limit(state_size_limit: Option<u64>) -> Self {
        Self { state_size_limit }
    }
}
```

That is, it seems the fact that the `DEFAULT_CONTRACT_STATE_SIZE_LIMIT` const lives here makes sense if the limit is not configurable. If it *is* configurable, `TrieViewer` doesn't need to know the default.

Reply:
I chose `new` because there's only one place override state_size_limit:
![image](https://user-images.githubusercontent.com/13259400/113913928-c40a6600-9791-11eb-8eb6-4e8a10a2d61a.png)

Reply:
>I chose new because there's only one place override state_size_limit:

uhu, but not how all *other* places are test code. The single non-test usage uses `new` immediately followed by `state_size_limit`. 

This is what makes me feel uneasy about the code: you would feel that `DEFAULT_CONTRACT_STATE_SIZE_LIMIT` is something that is used by default, but in the actual production code it is never used and is always overwritten by something else (although most likely it is overwritten with the same value).

Reply:
Addressed in ad5a057